pygfx/utils/show.py

```diff
@@ -11,56 +15,229 @@
+        self.renderer.render(self.scene, self.camera)
+
+        if self.after_render is not None:
+            self.after_render()
+
+        self.renderer.request_draw()
+
+    def show(
+        self,
+        object: WorldObject,
+        up=None,
+    ):
+        """Display a WorldObject
+
+        This function provides you with the basic scaffolding to visualize a given
+        WorldObject in a new window. While it does add scaffolding, it aims to be
+        fully customizable so that you can replace each piece as needed.
+
+        Parameters
+        ----------
+        object : gfx.WorldObject
+            The object to show. If it is not a :class:`gfx.Scene <pygfx.Scene>`
+            then Display will wrap it into a new scene containing lights and a
+            background.
+        up : gfx.Vector3
+            If set, and ``object`` does not contain a controller, set the camera
+            controller's up vector to this value.
 
-        background = Background(None, BackgroundMaterial((0, 1, 0, 1), (0, 1, 1, 1)))
-        scene.add(background)
+        Notes
+        -----
+        If you want to display multiple objects prefer :class:`gfx.Group
+        <pygfx.Group>` over :class:`gfx.Scene <pygfx.Scene>` to avoid the pitfall of
+        forgetting to add lights to the scene.
 
-    camera = PerspectiveCamera(70, 16 / 9)
-    look_at = camera.show_object(object)
-    scene.add(camera)
+        """
 
-    # Add lights if there are none
-    light_count = 0
+        if isinstance(object, Scene):
+            scene = object
+        else:
+            scene = Scene()
+            scene.add(object)
 
-    def check_light(ob):
-        nonlocal light_count
-        if isinstance(ob, Light):
-            light_count += 1
+            dark_gray = np.array((169, 167, 168, 255)) / 255
+            light_gray = np.array((100, 100, 100, 255)) / 255
 
-    scene.traverse(check_light, False)
-    if not light_count:
-        camera.add(DirectionalLight(0.8))
-        scene.add(AmbientLight(0.2))
+            background = Background(None, BackgroundMaterial(light_gray, dark_gray))
+            scene.add(background)
+            scene.add(AmbientLight(), DirectionalLight())
+        self.scene = scene
 
-    canvas = WgpuCanvas()
-    renderer = WgpuRenderer(canvas)
+        if not any(self.find_children(Light)):
+            warnings.warn(
+                "Your scene does not contain any lights. Some objects may not be visible"
+            )
 
-    controller = OrbitController(camera.position.clone(), look_at, up=up)
-    controller.add_default_event_handlers(renderer, camera)
+        existing_cameras = self.find_children(Camera)
+        if self.camera:
+            pass
+        elif any(existing_cameras):
+            self.camera = existing_cameras[0]
+        elif not self.camera:
+            self.camera = PerspectiveCamera(70, 16 / 9)
+            self.camera.add(DirectionalLight())
+            self.scene.add(self.camera)
```

Comment:
I saw that the two of you already had a discussion about this, but I think the docs are pretty clear that if a `Scene` is provided, that we don't add lights, so let's be consistent. @Korijn ?
```suggestion
            if object is not scene:
                self.camera.add(DirectionalLight())
                self.scene.add(self.camera)
```

Reply:
I will resolve https://github.com/pygfx/pygfx/pull/380#discussion_r1019023447 in favor of this discussion, since it is the same suggestion :)

Reply:
Yes, this is a good suggestion.

I would recommend to evaluate the condition in a single place though. For example you could put `add_lights = not isinstance(object, Scene)` near the top and re-use that condition in both places where lights are added.

Reply:
Refactored.